### ISSUE TITLE
[Android] CheckBox styles not changing with AppThemeBinding

### ISSUE THREAD

**Issue by rachelkang** [dotnet/maui#6253]
### Description

On Android, the CheckBox color / background color is not changing properly with the AppThemeBinding as per the following style"

```xaml
<Setter Property="Color" Value="{AppThemeBinding Light={StaticResource Primary}, Dark={StaticResource White}}" />
<Setter Property="BackgroundColor" Value="{AppThemeBinding Light={StaticResource White}, Dark={StaticResource Black}}" />
```

Using Background instead of BackgroundColor, and Brushes instead of Colors yield similar if not worse results. On certain occasions, in addition to failing to change styles, we see something like this:

![image](https://user-images.githubusercontent.com/21988533/164085777-72a0f0be-690b-42e9-9417-b570e245c5c1.png)

### Steps to Reproduce

1. Create a File > New .NET MAUI app
2. Add a CheckBox to the MainPage
3. Deploy the app to Android
4. Note the style of the CheckBox
5. Change the OS device theme
6. Return to app
7. Note the CheckBox style remains the same

### Version with bug

Release Candidate 1 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android API 32

**Comment by v-longmin** [comment]
Verified repro on vs 17.3.0 Preview 1.0 [32418.321.main]. Repro project:
[MauiApp10 (2).zip](https://github.com/dotnet/maui/files/8519606/MauiApp10.2.zip)
